atlasdb-cassandra-integration-tests/src/test/java/com/palantir/atlasdb/keyvalue/cassandra/CassandraTimestampBoundStoreTest.java

```diff
@@ -0,0 +1,141 @@
+/**
+ * Copyright 2017 Palantir Technologies
+ *
+ * Licensed under the BSD-3 License (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://opensource.org/licenses/BSD-3-Clause
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.palantir.atlasdb.keyvalue.cassandra;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.assertj.core.api.Assertions.assertThatThrownBy;
+
+import org.junit.After;
+import org.junit.ClassRule;
+import org.junit.Test;
+
+import com.google.common.collect.ImmutableMap;
+import com.palantir.atlasdb.AtlasDbConstants;
+import com.palantir.atlasdb.cassandra.CassandraKeyValueServiceConfigManager;
+import com.palantir.atlasdb.containers.CassandraContainer;
+import com.palantir.atlasdb.containers.Containers;
+import com.palantir.atlasdb.encoding.PtBytes;
+import com.palantir.atlasdb.keyvalue.api.Cell;
+import com.palantir.atlasdb.timestamp.AbstractDbTimestampBoundStoreTest;
+import com.palantir.timestamp.MultipleRunningTimestampServiceError;
+import com.palantir.timestamp.TimestampBoundStore;
+
+public class CassandraTimestampBoundStoreTest extends AbstractDbTimestampBoundStoreTest {
+    private static final long CASSANDRA_TIMESTAMP = 0L;
+    private static final String ROW_AND_COLUMN_NAME = "ts";
+    public static final long NEW_LIMIT = 100;
+
+    @ClassRule
+    public static final Containers CONTAINERS = new Containers(CassandraTimestampIntegrationTest.class)
+            .with(new CassandraContainer());
+
+    private CassandraKeyValueService kv = CassandraKeyValueService.create(
+            CassandraKeyValueServiceConfigManager.createSimpleManager(CassandraContainer.KVS_CONFIG),
+            CassandraContainer.LEADER_CONFIG);
+
+    @Override
+    public TimestampBoundStore createTimestampBoundStore() {
+        return CassandraTimestampBoundStore.create(kv);
+    }
+
+    @Test
+    public void storeWithEmptyTableThrows() {
+        assertThatThrownBy(() -> store.storeUpperLimit(NEW_LIMIT + 1))
+                .isExactlyInstanceOf(Error.class);
+    }
+
+    @Test
+    public void canGetNewFormat() {
+        insertTimestampWithCorrectId(NEW_LIMIT + 1);
+        assertCorrect();
+    }
+
+    @Test
+    public void canGetOldFormat() {
+        insertTimestampOld(NEW_LIMIT + 1);
+        assertCorrect();
+    }
+
+    @Test
+    public void storeWithWrongTimestampCorrectIdSucceeds() {
+        long limit = store.getUpperLimit();
+        insertTimestampWithCorrectId(NEW_LIMIT);
+        assertThat(limit).isNotEqualTo(NEW_LIMIT);
```

Comment:
If possible, there should be a message attached to this assert - if it breaks, then something is probably wrong with the test, or with something other than what the test is actually testing.

This line happens several times in this file - you might find it convenient to extract a method before adding a message.